client/src/connection/itc/index.ts

```diff
@@ -0,0 +1,387 @@
+      this._runResolve = resolve;
+      this._runReject = reject;
+    });
+
+    if (this._shellProcess && !this._shellProcess.killed) {
+      this._runResolve();
+      return; // manually terminate to avoid executing the code below
+    }
+
+    this._shellProcess = spawn("powershell.exe /nologo -Command -", {
+      shell: true,
+      env: process.env,
+    });
+    this._shellProcess.stdout.on("data", this.onShellStdOut);
+    this._shellProcess.stderr.on("data", this.onShellStdErr);
+    this._shellProcess.stdin.write(scriptContent + "\n", this.onWriteComplete);
+    this._shellProcess.stdin.write(
+      "$runner = New-Object -TypeName SASRunner\n",
+      this.onWriteComplete,
+    );
+
+    /*
+     * There are cases where the higher level run command will invoke setup multiple times.
+     * Avoid re-initializing the session when this happens. In a first run scenario a work dir
+     * will not exist. The work dir should only be deleted when close is invoked.
+     */
+    if (!this._workDirectory) {
+      const { host, port, protocol, username } = this._config;
+      this._shellProcess.stdin.write(`$profileHost = "${host}"\n`);
+      this._shellProcess.stdin.write(`$port = ${port}\n`);
+      this._shellProcess.stdin.write(`$protocol = ${protocol}\n`);
+      this._shellProcess.stdin.write(`$username = "${username}"\n`);
+      const password = await this.fetchPassword();
+      this._shellProcess.stdin.write(`$password = "${password}"\n`);
+      this._shellProcess.stdin.write(
+        `$serverName = "${
+          protocol === ITCProtocol.COM ? "ITC Local" : "ITC IOM Bridge"
+        }"\n`,
+      );
+      this._shellProcess.stdin.write(
+        `$runner.Setup($profileHost,$username,$password,$port,$protocol,$serverName)\n`,
+        this.onWriteComplete,
+      );
+      this._shellProcess.stdin.write(
+        "$runner.ResolveSystemVars()\n",
+        this.onWriteComplete,
+      );
+
+      if (this._config.sasOptions?.length > 0) {
+        const sasOptsInput = `$sasOpts=${this.formatSASOptions(
+          this._config.sasOptions,
+        )}\n`;
+        this._shellProcess.stdin.write(sasOptsInput, this.onWriteComplete);
+        this._shellProcess.stdin.write(
+          `$runner.SetOptions($sasOpts)\n`,
+          this.onWriteComplete,
+        );
+      }
+    }
+
+    // free objects in the scripting env
+    process.on("exit", async () => {
+      close();
+    });
+
+    return setupPromise;
+  };
+
+  private storePassword = async () =>
+    await setSecret(PASSWORD_KEY, this._password);
+
+  private clearPassword = async () => {
+    await setSecret(PASSWORD_KEY, "");
+    this._password = "";
+  };
+
+  private fetchPassword = async (): Promise<string> => {
+    if (this._config.protocol === ITCProtocol.COM) {
+      return "";
+    }
+
+    const storedPassword = await getSecret(PASSWORD_KEY);
+    if (storedPassword) {
+      return storedPassword;
```

Comment:
You're not setting `this._password` here, so `this._password` can still be `""` right?
When `storePassword()` called, `""` will be stored which is not expected, right?

Reply:
Resolved in https://github.com/sassoftware/vscode-sas-extension/pull/592/commits/4637097276508551fd84e58a5bc7078ea5423581